pkg/testutil/sets.go

```diff
@@ -0,0 +1,54 @@
+package testutil
+
+type Set[K comparable] struct {
+	entities map[K]any
+}
+
+func NewSetFromMap[K comparable, V any](m map[K]V) *Set[K] {
+	entities := make(map[K]any, 0)
+	for k := range m {
+		entities[k] = true
+	}
+	return &Set[K]{
```

Comment:
This file should have more new lines.

Reply:
fixed in lint commit

Reply:
I appreciate that! Getting the minor lint out of the way makes it a lot easier for me to focus on the logic.